### PR TITLE
Add PNSE breaking change for strong-name APIs

### LINKED ISSUE #29681: [Breaking change]: Throwing PNSE for some APIs that was no-op previously
### Strong name signing is not supported and throws PlatformNotSupportedException

Related to https://github.com/dotnet/runtime/pull/50941 and https://github.com/dotnet/runtime/issues/50529

As part of [Annotating unsupported APIs in System.Reflection](https://github.com/dotnet/runtime/issues/50529) we found few APIs that are not supported in .NetCore and also not doing anything when accessed. We changed them to throw PNSE with https://github.com/dotnet/runtime/pull/50941/files as the API usage will eventually throw PNSE at runtime when they are used for a real.

This change has made in .Net 6.0, affected APIS:

| API | Previous Behavior | New behavior |
| --- | --- | --- |
[AssemblyName.KeyPair](https://docs.microsoft.com/en-us/dotnet/api/system.reflection.assemblyname.keypair?view=net-6.0) | No-op | throws PlatformNotSupportedException
[StrongNameKeyPair(FileStream keyPairFile)](https://docs.microsoft.com/en-us/dotnet/api/system.reflection.strongnamekeypair.-ctor?view=net-6.0#system-reflection-strongnamekeypair-ctor(system-io-filestream)) | Reads the stream but does nothing | throws PlatformNotSupportedException
[StrongNameKeyPair(byte[] keyPairArray)](https://docs.microsoft.com/en-us/dotnet/api/system.reflection.strongnamekeypair.-ctor?view=net-6.0#system-reflection-strongnamekeypair-ctor(system-byte())) | No-op | throws PlatformNotSupportedException

### Version

.Net core 6.0

### Previous behavior

No-op

### New behavior

Throws PlatformNotSupportedException

### Type of breaking change

- [x] **Binary incompatible**: Existing binaries may encounter a breaking change in behavior, such as failure to load/execute or different run-time behavior.
- [ ] **Source incompatible**: Source code may encounter a breaking change in behavior when targeting the new runtime/component/SDK, such as compile errors or different run-time behavior.

### Reason for change

An application that uses the API would compile and ran but only if it is not used. But as soon as if the instance is used in any code path, it would throw a runtime exception, most likely PNSE. So to make more explicit that this scenario was unsupported, the exception-throwing logic was moved into the instance constructor and since no instances could ever be created, propagated to public entry points returning this type (like the `AssemblyName.KeyPair`).

### Recommended action

 Strong name signing is not supported in .Net Core completely, there is no workaround. With the PR we have obsoleted this type and related APIs

### Feature area

Core .NET libraries

### Affected APIs

[AssemblyName.KeyPair](https://docs.microsoft.com/en-us/dotnet/api/system.reflection.assemblyname.keypair?view=net-6.0) 
[StrongNameKeyPair(FileStream keyPairFile)](https://docs.microsoft.com/en-us/dotnet/api/system.reflection.strongnamekeypair.-ctor?view=net-6.0#system-reflection-strongnamekeypair-ctor(system-io-filestream)) 
[StrongNameKeyPair(byte[] keyPairArray)](https://docs.microsoft.com/en-us/dotnet/api/system.reflection.strongnamekeypair.-ctor?view=net-6.0#system-reflection-strongnamekeypair-ctor(system-byte()))

### PR DESCRIPTION
**Pull request by gewarren**
Fixes #29681

[Preview link](https://review.docs.microsoft.com/en-us/dotnet/core/compatibility/core-libraries/6.0/strong-name-signing-exceptions?branch=pr-en-us-29686)

### CODE REVIEW

File: docs/core/compatibility/core-libraries/6.0/strong-name-signing-exceptions.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+---
+title: "Breaking change: Strong-name APIS throw PlatformNotSupportedException"
+description: Learn about the .NET 6 breaking change in core .NET libraries where an exception is thrown in StrongNameKeyPair constructors and AssemblyName.KeyPair.
+ms.date: 05/31/2022
+---
+# Strong-name APIS throw PlatformNotSupportedException
+
+A few [APIs](#affected-apis) that aren't supported in .NET/.NET Core but didn't doing anything when accessed have been changed to now throw a <xref:System.PlatformNotSupportedException> at run time. Previously, using these APIs would eventually result in a run-time exception further along; the exception is now thrown when the type is instantiated or first accessed.
```
Comment (KalleOlaviNiemitalo) [inline docs/core/compatibility/core-libraries/6.0/strong-name-signing-exceptions.md:8]: "didn't doing" seems ungrammatical.

File: docs/core/compatibility/6.0.md [hunk 2]
```diff
@@ -73,6 +73,7 @@ If you're migrating an app to .NET 6, the breaking changes listed here might aff
 | [Standard numeric format parsing precision](core-libraries/6.0/numeric-format-parsing-handles-higher-precision.md) | ✔️ | ❌ | Preview 2 |
 | [Static abstract members in interfaces](core-libraries/6.0/static-abstract-interface-methods.md) | ❌ | ✔️ | Preview 7 |
 | [StringBuilder.Append overloads and evaluation order](core-libraries/6.0/stringbuilder-append-evaluation-order.md) | ❌ | ✔️ | RC 1 |
+| [Strong-name APIS throw PlatformNotSupportedException](core-libraries/6.0/strong-name-signing-exceptions.md) | ❌ | ✔️ | |
```
Comment (KalleOlaviNiemitalo) [inline docs/core/compatibility/6.0.md:76]: That's capitalised as "APIs" elsewhere. The ToC additions also use "APIS".

**Review by KalleOlaviNiemitalo — CHANGES_REQUESTED** [review]

**Comment by gewarren** [comment]
Thanks @KalleOlaviNiemitalo!

File: docs/core/compatibility/6.0.md [hunk 3]
```diff
@@ -73,6 +73,7 @@ If you're migrating an app to .NET 6, the breaking changes listed here might aff
 | [Standard numeric format parsing precision](core-libraries/6.0/numeric-format-parsing-handles-higher-precision.md) | ✔️ | ❌ | Preview 2 |
 | [Static abstract members in interfaces](core-libraries/6.0/static-abstract-interface-methods.md) | ❌ | ✔️ | Preview 7 |
 | [StringBuilder.Append overloads and evaluation order](core-libraries/6.0/stringbuilder-append-evaluation-order.md) | ❌ | ✔️ | RC 1 |
+| [Strong-name APIs throw PlatformNotSupportedException](core-libraries/6.0/strong-name-signing-exceptions.md) | ❌ | ✔️ | |
```
Comment (buyaa-n) [inline docs/core/compatibility/6.0.md:76]: NIT: it was updated in preview 4
```suggestion
| [Strong-name APIs throw PlatformNotSupportedException](core-libraries/6.0/strong-name-signing-exceptions.md) | ❌ | ✔️ |  Preview 4 |
```

Comment (buyaa-n) [inline docs/core/compatibility/6.0.md:76]: NIT: it was updated in preview 4
```suggestion
| [Strong-name APIs throw PlatformNotSupportedException](core-libraries/6.0/strong-name-signing-exceptions.md) | ❌ | ✔️ |  Preview 4 |
```

**Review by buyaa-n — APPROVED** [review]
LGTM, thanks!

**Pull request merged**